src/core/api/thread_api.cpp

```diff
@@ -202,6 +202,12 @@ otError otThreadSetNetworkName(otInstance *aInstance, const char *aNetworkName)
 
     VerifyOrExit(AsCoreType(aInstance).Get<Mle::MleRouter>().IsDisabled(), error = kErrorInvalidState);
 
+#if !OPENTHREAD_CONFIG_NETWORK_NAME_ZERO_LENGTH_ENABLE
```

Comment:
We need to define and document this config in one of the headers in `src/core/config` folder (set the default value). Maybe in `src/core/config/misc.h` or `src/core/config/mac.h`?

Also suggest maybe renaming it to `OPENTHREAD_CONFIG_ALLOW_EMPTY_NETWORK_NAME`? (I feel `ENABLE` term may make sense for bigger features and/or behavior change).